Storage/src/Connection/Rest.php

```diff
@@ -600,4 +602,30 @@ protected function supportsBuiltinCrc32c()
     {
         return Builtin::supports(CRC32::CASTAGNOLI);
     }
+
+    /**
+     * Add the required retry function and send the request.
+     *
+     * @param string $resource resource name, eg: buckets.
+     * @param string $method method name, eg: get
+     * @param array $args
+     */
+    private function sendWithRetry($resource, $method, $args)
+    {
+        $retryMap = [
+            'projects.resources.serviceAccount' => 'serviceaccount',
+            'projects.resources.hmacKeys' => 'hmacKey',
+            'bucketAccessControls' => 'bucket_acl',
+            'defaultObjectAccessControls' => 'default_object_acl',
+            'objectAccessControls' => 'object_acl'
+        ];
+        $retryResource = isset($retryMap[$resource]) ? $retryMap[$resource] : $resource;
+        $args['restRetryFunction'] = $this->getRestRetryFunction(
```

Comment:
This will override any custom restRetryFunction supplied by the user. So we should do this by checking `if not isset($args['restRetryFunction']` and only then assigning it's value.

Reply:
This check is available in the getRestRetryFunction itself. Please see https://github.com/googleapis/google-cloud-php/pull/5709/files/926dc07dd381eca22951db1c8cadcba72d5f0472#diff-1c3cf20d08267cf9f283261e1f3ae580331c768f17676e28288d56bf4a14be27R98

Reply:
There are two ways a user can supply the retry function, either through the constructor of the class or through an option at the individual method call. I think @yash30201's comment refers to the latter, could we make sure we cover the former as well?

Reply:
So, it's not documented but I think we can do this:
```
$client = new StorageClient([
    'restRetryFunction' => function() {}
]);
```
This can be observed as the `$config` is forwarded from `StorageClient` -> `Rest` -> `RequestWrapper` constructors.

Clearly this won't be honoured from our current application. So, along with checking if a user has supplied a retry function in config, we need to also check if a user had supplied the retry function during initialisation of the `StorageClient`.

Reply:
Okay, Will provide a way take care of this condition. Also, will add this option to the docblock

Reply:
Changed the code to accommodate this change, Thank you!

Reply:
> @dwsupplee Could you please tell me which constructor you are talking about?
> 
> I have checked the StroageClient constructor and the available arguments are the following:
> 
> ```
> /**
>  * Create a Storage client.
>  *
>  * @param array $config [optional] {
>  *     Configuration options.
>  *
>  *     @type string $apiEndpoint The hostname with optional port to use in
>  *           place of the default service endpoint. Example:
>  *           `foobar.com` or `foobar.com:1234`.
>  *     @type string $projectId The project ID from the Google Developer's
>  *           Console.
>  *     @type CacheItemPoolInterface $authCache A cache used storing access
>  *           tokens. **Defaults to** a simple in memory implementation.
>  *     @type array $authCacheOptions Cache configuration options.
>  *     @type callable $authHttpHandler A handler used to deliver Psr7
>  *           requests specifically for authentication.
>  *     @type FetchAuthTokenInterface $credentialsFetcher A credentials
>  *           fetcher instance.
>  *     @type callable $httpHandler A handler used to deliver Psr7 requests.
>  *           Only valid for requests sent over REST.
>  *     @type array $keyFile The contents of the service account credentials
>  *           .json file retrieved from the Google Developer's Console.
>  *           Ex: `json_decode(file_get_contents($path), true)`.
>  *     @type string $keyFilePath The full path to your service account
>  *           credentials .json file retrieved from the Google Developers
>  *           Console.
>  *     @type float $requestTimeout Seconds to wait before timing out the
>  *           request. **Defaults to** `0` with REST and `60` with gRPC.
>  *     @type int $retries Number of retries for a failed request.
>  *           **Defaults to** `3`.
>  *     @type array $scopes Scopes to be used for the request.
>  *     @type string $quotaProject Specifies a user project to bill for
>  *           access charges associated with the request.
>  * }
>  */
>  ```
>  
>  Please let me know what am I missing here?
>  
>  
>  Other comments has been resolved, thank you!
> ```

Done the required changes to accommodate this.